sycl/include/CL/sycl/detail/device_filter.hpp

```diff
@@ -0,0 +1,96 @@
+//==---------- device_filter.hpp - SYCL device filter descriptor -----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/backend_types.hpp>
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/info/info_desc.hpp>
+
+#include <iostream>
+#include <string>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+struct device_filter {
+  backend Backend = backend::host;
+  info::device_type DeviceType = info::device_type::all;
+  int DeviceNum = 0;
+  bool HasBackend = false;
+  bool HasDeviceType = false;
+  bool HasDeviceNum = false;
```

Comment:
Do we need device ID here. As for my understanding enumeration happens by a platform. And we have single device per platform.
@mlychkov Can we have more that one FPGA device in FPGA platform?

Reply:
Yes, it is possible configuration with several devices on one FPGA platform. CL_CONFIG_CPU_EMULATE_DEVICES env var can be used to setup configuration with several FPGA emulator devices on FPGA platform.